src/Parser/Desugar.ml

```diff
@@ -708,7 +708,11 @@ and tr_def ?(public=false) (def : Raw.def) =
     let (pub_type, pub_ctors) =
       match vis with
       | DV_Private  -> (public, public)
-      | DV_Abstract -> (true,   public)
+      | DV_Abstract -> if public then
+                       (InterpLib.Error.report ~pos:def.pos
+                       ~cls:InterpLib.Error.Warning
+                       "'abstr' data modifier inside public rec block");
```

Comment:
The message string should be improved, because of the two following reasons.
1. It doesn't help the programmer, unless he knows the language very well.
2. It would scale poorly to nested sections (grouping several definitions that share local `implicit` declarations).

My suggestion is to say something like "This 'abstr' data modifier has no effect. It is overridden by 'public' modifier of entire group of definitions."